sys/include/net/conn/udp.h

```diff
@@ -51,6 +51,9 @@ typedef struct conn_udp conn_udp_t;
  * @param[in] family    The family of @p addr (see @ref net_af).
  * @param[in] port      The local UDP port for @p conn.
  *
+ * @note    @ref conn_udp_recvfrom needs to be called from the same thread as
+ *          for this function.
+ *
```

Comment:
Just mention GNRC in the text. `conn` is conceptualized as *general* connection API. So regardless if other implementations exist or not: this is the current state in *GNRC*, and not an issue of this API.